src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1211,9 +1211,21 @@ public static string GetModulePath(string currentProcessModulePath, string hklmM
                 // sharedModulePath
                 // systemModulePath
                 currentProcessModulePath = AddToPath(currentProcessModulePath, personalModulePathToUse, 0);
-                int insertIndex = PathContainsSubstring(currentProcessModulePath, personalModulePathToUse) + personalModulePathToUse.Length + 1;
+                int insertIndex = currentProcessModulePath.IndexOf(Path.PathSeparator, PathContainsSubstring(currentProcessModulePath, personalModulePathToUse));
```

Comment:
Does the issue also occur when the alternate path separator is used? By this I include `\` on Unix, even though that isn't `Path.AlternatePathSeparator` there

Reply:
@rjmholt I think you're thinking of `AlternateDirectorySeparatorChar`; I don't think there is an `AlternatePathSeparator` (`:` on Unix, `;` on Windows). As long as we're targeting the path separator itself, we should be OK here 🙂

Reply:
Oh, yep, got those confused. Read the "sub-path trailing separator" as "trailing directory separator" rather than "trailing path separator"

Reply:
Every. Single. Time. 

I _wish_ they'd rename them, I don't know a single person who _hasn't_ confused them at least a dozen times. 😂